### PR TITLE
update svg drawing order of bar chart to: grid -> series -> labels

### PR DESCRIPTION
**Pull request by alexZielonko**
https://github.com/gionkunz/chartist-js/issues/306

### CODE REVIEW

File: dist/chartist.js [hunk 1]
```diff
@@ -3525,6 +3530,8 @@ var Chartist = {
     showLabel: true,
     // Label position offset from the standard position which is half distance of the radius. This value can be either positive or negative. Positive values will position the label away from the center.
     labelOffset: 0,
+    // This option can be set to 'inside', 'outside' or 'center'. Positioned with 'inside' the labels will be placed on half the distance of the radius to the border of the Pie by respecting the 'labelOffset'. The 'outside' option will place the labels at the border of the pie and 'center' will place the labels in the absolute center point of the chart. The 'center' option only makes sense in conjunction with the 'labelOffset' option.
+    labelPosition: 'inside',
```
Comment (gionkunz) [inline dist/chartist.js]: Did you rebase before creating the PR? Why is this change in your PR?

Comment (gionkunz) [inline dist/chartist.js]: Ahh.. sorry this is the dist file. Please never add changes to dist files in PR's. Can you exclude them and resubmit your PR? Thanks!

**Comment by gionkunz** [comment]
Shouldn't this change also be done to the Line chart?

**Comment by alexZielonko** [comment]
I reverted the 'built project' commit and added the same functionality to the line chart. Let me know if any other changes are necessary.

**Comment by gionkunz** [comment]
Hey there. Sorry for my slow pace but I didn't had a chance to work on chartist lately. I just checkout your branch and tested a bit. One thing I've noticed is that due to moving the grid and label creation to the end, also the order of the events changed. So no first the series (line, point, area) events trigger and then the label and grid events trigger. This is not ideal as I believe grid and label events should trigger first.

Can you check if you could leave the order as before but use the insertFirst parameter in the [Svg.elem function](http://localhost:9000/api-documentation.html#chartistsvg-function-elem)? This way you could just change the order how the group elements are placed into the SVG element which would affect appearance but not the structural sequence how the chart is created.

**Comment by gionkunz** [comment]
e.g. `var labelGroup = this.svg.elem('g', null, options.classNames.labelGroup, true);`

**Comment by gionkunz** [comment]
Actually best would be to have all groups created on top of the create function where the order is how they should appear as groups in the SVG element.

**Comment by gionkunz** [comment]
@missedtheboat I'm working on some changes to make Axis Type user configurable and I was incorporating your suggested drawing order together with this change. Thanks for your contribution, it's really appreciated!

**Pull request closed without merging**